lecture14/pageobjects/page_objects.py

```diff
@@ -0,0 +1,112 @@
+import time
+from typing import List
+
+from selenium.common.exceptions import NoSuchElementException
+from selenium.webdriver.chrome.webdriver import WebDriver
+from selenium.webdriver.support import expected_conditions as EC
+from selenium.webdriver.support.ui import WebDriverWait
+
+from lecture14.models.product import Product
+from lecture14.models.user import User
+from lecture14.pageobjects.selectors import *
+
+
+class MainPage:
+    url = "https://www.demoblaze.com/"
+
+    def __init__(self, driver: WebDriver):
+        self.driver = driver
+
+    def click_product_by_title(self, title):
+        product_titles = self.driver.find_elements(*MainPageSelectors.PRODUCTS_TITLES)
+        matching_titles = [pt for pt in product_titles if pt.text == title]
+        if matching_titles:
+            matching_titles[0].click()
+        else:
+            raise NoSuchElementException(f'Element with title {title} was not found')
+
+    def click_monitors_category(self):
+        WebDriverWait(self.driver, 10).until(
+            EC.visibility_of_element_located(MainPageSelectors.MONITORS_CATEGORY)).click()
+
+    def get_product_with_highest_price(self) -> Product:
+        time.sleep(3)

```

Comment:
it is better to avoid sleep if possible
in case there is no sign for event to wait for there should be a doc string with explanation

Reply:
Added comment.